### ISSUE TITLE
Overloaded dunder method choosing wrong overload

### ISSUE THREAD

**Issue by adamjstewart** [astral-sh/ty#3323]
### Summary

This is admittedly an extremely convoluted bug, so please tell me if the actual bug is in pandas, not ty.

The following code:
```python
from pandas import Timedelta

t1 = Timedelta(days=5)
t2 = Timedelta(days=1)
x = t1 / t2
reveal_type(x)
```
Shows that `ty` considers the output type to be Timedelta. However, according to [pandas-stubs](https://github.com/pandas-dev/pandas-stubs/blob/v3.0.0.260204/pandas-stubs/_libs/tslibs/timedeltas.pyi#L242):
```python
class Timedelta(timedelta):
    ...
    # Override due to more types supported than timedelta
    @overload  # type: ignore[override]
    # pyrefly: ignore[bad-override]
    def __truediv__(self, other: Just[int] | Just[float]) -> Self: ...
    @overload
    def __truediv__(self, other: Self | NaTType) -> float: ...
    @overload
    def __truediv__(
        self, other: np_ndarray[ShapeT, np.integer | np.floating]
    ) -> np_ndarray_td[ShapeT]: ...
    @overload
    def __truediv__(  # ty: ignore[invalid-method-override]
        self, other: np_ndarray_td[ShapeT]
    ) -> np_ndarray_float[ShapeT]: ...
```
the correct type should be float. Printing the value confirms this.

Is this a bug in ty overload handling, or dunder handling, or due to the lack of a non-overloaded version, or something else? I tried removing all comments and it isn't due to the comment between the `@overload` and function signature.

### Version

ty 0.0.29 (438a78d68 2026-04-05)

**Comment by sharkdp** [comment]
The problematic overload is this one:
```py
    @overload  # type: ignore[override]
    # pyrefly: ignore[bad-override]
    def __truediv__(self, other: Just[int] | Just[float]) -> Self: ...
```

ty thinks that `Timedelta` is assignable to `Just[int] | Just[float]`. That protocol is defined as

```py
class Just(Protocol, Generic[T]):
    @property  # type: ignore[override]
    @override
    def __class__(self, /) -> type[T]: ...  # pyrefly: ignore[bad-override]
    @__class__.setter
    @override
    def __class__(self, t: type[T], /) -> None: ...
```

Here's a self-contained example that shows the same problem ([Playground](https://play.ty.dev/edccbf87-df34-4ee1-baea-3379e174f809)):

```py
from typing import Protocol, Generic, TypeVar, override

T = TypeVar("T")

class Just(Protocol, Generic[T]):
    @property  # type: ignore[override]
    @override
    def __class__(self, /) -> type[T]: ...  # pyrefly: ignore[bad-override]
    @__class__.setter
    @override
    def __class__(self, t: type[T], /) -> None: ...


class Whatever: ...

x: Just[float] = Whatever()
```

Other type checkers reject this assignment.

**Comment by adamjstewart** [comment]
Thanks for digging into this! I can confirm that removing `Just` results in the correct behavior. It looks like this was added in https://github.com/pandas-dev/pandas-stubs/pull/1421.

So does this mean there is a bug in the `Just` implementation, or a bug in ty? I'll fully admit I have no idea how `Just` is supposed to work.

**Comment by sharkdp** [comment]
No, I think this is a problem in ty. I will link #1379 as a parent issue.

**Comment by carljm** [comment]
This is now fixed (in https://github.com/astral-sh/ruff/pull/25332)
